Error when signtool.exe can't be found should include the path we looked in

[This line](https://github.com/Microsoft/msbuild/blob/98d38cbf34fe35142792556aff80cec08d7315cc/src/XMakeTasks/ManifestUtil/SecurityUtil.cs#L773) makes it hard to figure out *why* signtool appeared to be missing. We had an internal bug where the .exe was actually not present on disk, but the error was just:

```
1>Task "SignFile" (TaskId:43)
1>  Task Parameter:CertificateThumbprint=AAD0491335D003B4BAE60CCCEE28EC62600D224D (TaskId:43)
1>  Task Parameter:SigningTarget=bin\Debug\app.publish\\setup.exe (TaskId:43)
1>C:\Program Files (x86)\MSBuild\Microsoft\VisualStudio\v14.0\OfficeTools\Microsoft.VisualStudio.Tools.Office.targets(609,9): error MSB3482: An error occurred while signing: SignTool.exe not found.
1>Done executing task "SignFile" -- FAILED. (TaskId:43)
```

Would have been nice to know where it was looking!

--- a/src/XMakeTasks/ManifestUtil/SecurityUtil.cs
+++ b/src/XMakeTasks/ManifestUtil/SecurityUtil.cs
@@ -703,9 +703,7 @@ namespace Microsoft.Build.Tasks.Deployment.ManifestUtilities
 
         private static void SignPEFile(X509Certificate2 cert, System.Uri timestampUrl, string path, System.Resources.ResourceManager resources, bool useSha256)
         {
-            if (GetPathToTool() == null) throw new ApplicationException(resources.GetString("SecurityUtil.SigntoolNotFound"));
-
-            ProcessStartInfo startInfo = new ProcessStartInfo(GetPathToTool(), GetCommandLineParameters(cert.Thumbprint, timestampUrl, path, useSha256));
+            ProcessStartInfo startInfo = new ProcessStartInfo(GetPathToTool(resources), GetCommandLineParameters(cert.Thumbprint, timestampUrl, path, useSha256));
             startInfo.CreateNoWindow = true;
             startInfo.UseShellExecute = false;
             startInfo.RedirectStandardError = true;
@@ -760,7 +758,7 @@ namespace Microsoft.Build.Tasks.Deployment.ManifestUtilities
             return commandLine.ToString();
         }
 
-        internal static string GetPathToTool()
+        internal static string GetPathToTool(System.Resources.ResourceManager resources)
         {
 #pragma warning disable 618 // Disabling warning on using internal ToolLocationHelper API. At some point we should migrate this.
             string toolPath = ToolLocationHelper.GetPathToWindowsSdkFile(ToolName, TargetDotNetFrameworkVersion.VersionLatest, VisualStudioVersion.VersionLatest);
@@ -771,7 +769,7 @@ namespace Microsoft.Build.Tasks.Deployment.ManifestUtilities
             if (toolPath == null)
                 toolPath = Path.Combine(Directory.GetCurrentDirectory(), ToolName);
             if (!File.Exists(toolPath))
-                toolPath = null;
+                throw new ApplicationException(String.Format(CultureInfo.CurrentCulture, resources.GetString("SecurityUtil.SigntoolNotFound"), toolPath));
             return toolPath;
 #pragma warning restore 618
         }

--- a/src/XMakeTasks/ManifestUtil/Strings.resx
+++ b/src/XMakeTasks/ManifestUtil/Strings.resx
@@ -145,7 +145,7 @@
     <value xml:space="preserve">Warning while signing {0}. {1}</value>
   </data>
   <data name="SecurityUtil.SigntoolNotFound">
-    <value xml:space="preserve">SignTool.exe not found.</value>
+    <value xml:space="preserve">SignTool.exe was not found at path {0}.</value>
   </data>
   <data name="SecurityUtil.TimestampUrlNotFound">
     <value xml:space="preserve">Timestamp URL server name or address could not be resolved.</value>

